### PR TITLE
[ui] Display computation time for "running" or "finished" nodes

### PR DESCRIPTION
**Pull request by cbentejac**
## Description

This PR displays, for a node whose status is either "FINISHED" or "RUNNING", the computation time in the title bar of the node. For nodes with several chunks, the displayed time corresponds to the cumulated computation time of every chunk; in that case, the computation time for the longest chunk is also displayed in a tooltip.

### CODE REVIEW

File: meshroom/ui/qml/GraphEditor/NodeEditor.qml [hunk 1]
```diff
@@ -24,6 +24,90 @@ Panel {
...
+                    if (node.elapsedTime > 0) { // At least a chunk is done running
+                        return "Running for: " + getTimeStr(node.elapsedTime)
+                    } else {
+                        return (node.chunks.count > 1) ? "First chunk running" : "Node running"
+                    }
+                } else if (node.isFinishedOrRunning()) {
+                    /* Either all chunks finished running or the last one is running
+                        * Placed inside an "else if" instead of "else" to avoid entering the functions
+                        * when there is no real use */
+                    return "Ran for: " + getTimeStr(node.elapsedTime)
```
Comment (fabiencastan) [inline meshroom/ui/qml/GraphEditor/NodeEditor.qml:40]:
```suggestion
                    return getTimeStr(node.elapsedTime)
```

File: meshroom/ui/qml/GraphEditor/NodeEditor.qml [hunk 2]
```diff
@@ -24,6 +24,90 @@ Panel {
...
+            visible: {
+                if (node !== null) {
+                    if ((node.isFinishedOrRunning() || node.isSubmittedOrRunning())) {
+                        return true
+                    }
+                }
+                return false
+            }
+
+            ToolTip.text: {
```
Comment (fabiencastan) [inline meshroom/ui/qml/GraphEditor/NodeEditor.qml:54]: I didn't manage to see the tooltip.

File: meshroom/ui/qml/GraphEditor/NodeEditor.qml (hunk 1 above)
Comment (cbentejac) [reply meshroom/ui/qml/GraphEditor/NodeEditor.qml]: Done

File: meshroom/ui/qml/GraphEditor/NodeEditor.qml [hunk 3]
```diff
@@ -24,6 +24,92 @@ Panel {
...
+                    elapsed = elapsed - (hours * 3600)
+                    finalTime += hours + "h"
+                }
+                if (elapsed > 60) {
+                    min = Math.floor(elapsed / 60)
+                    elapsed = elapsed - (min * 60)
+                    finalTime += min + "m"
+                }
+
+                finalTime += Number(elapsed.toLocaleString(Qt.locale('en-US'))) + "s"
```
Comment (fabiencastan) [inline meshroom/ui/qml/GraphEditor/NodeEditor.qml:95]: Need a round to convert to interger.

"1h52m3.12s" is quite difficult to read (not coherent int/float between units)
And second precision is enough, so we should move to "1h52m3s".

File: meshroom/ui/qml/GraphEditor/NodeEditor.qml (hunk 2 above)
Comment (fabiencastan) [reply meshroom/ui/qml/GraphEditor/NodeEditor.qml:56]: Oh it's only for chunks, so that's good.

File: meshroom/ui/qml/GraphEditor/NodeEditor.qml [hunk 4]
```diff
@@ -24,6 +24,92 @@ Panel {
...
+                    }
+                }
+                return false
+            }
+
+            ToolTip.text: {
+                if (node !== null && (node.isFinishedOrRunning() || (node.isSubmittedOrRunning() && node.elapsedTime > 0))) {
+                    var longestChunkTime = getLongestChunkTime(node.chunks)
+                    if (longestChunkTime > 0)
+                        return "(longest chunk: " + getTimeStr(longestChunkTime) + ")"
```
Comment (fabiencastan) [inline meshroom/ui/qml/GraphEditor/NodeEditor.qml:60]: No need for ( ) as it's the only message.
```suggestion
                        return "Longest chunk: " + getTimeStr(longestChunkTime)
```

Comment (fabiencastan) [reply meshroom/ui/qml/GraphEditor/NodeEditor.qml:60]: Maybe good to display the number of chunks.

Comment (cbentejac) [reply meshroom/ui/qml/GraphEditor/NodeEditor.qml]: Done (and number of chunks added)

**Review by fabiencastan — APPROVED** [review]

**Pull request merged**